src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -759,6 +766,39 @@ PurgeConnectionForPlacement(ShardPlacement *placement)
 
 
 /*
+ * Removes a given connection from the transaction participant hash, based on
+ * the host and port of the provided connection. If the hash is not NULL, it
+ * MUST contain the provided connection, or a FATAL error is raised.
+ */
```

Comment:
If you want to make me really happy, you could add a reference to PurgeConnectionForPlacement.